server/services/analyticsService.ts

```diff
@@ -141,4 +151,76 @@ export default class AnalyticsService {
     })
     return { levels, report: prisonersOnLevels, lastUpdated: new Date(), dataSource: 'NOMIS' }
   }
+
+  async getIncentiveLevelsByEthnicity(
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    prison: string
+  ): Promise<{ levels: string[] } & Report<PrisonersOnLevelsByEthnicity[]>> {
+    // TODO: fake response; move into test
+    const levels = ['Basic', 'Standard', 'Enhanced', 'Enhanced 2']
+    const response: [string, number, number, number, number][] = [
+      ['Asian or Asian British', 3, 39, 42, 0],
+      ['Black or Black British', 8, 62, 46, 2],
+      ['Mixed', 2, 30, 40, 0],
+      ['Other', 1, 7, 8, 0],
+      ['White', 28, 615, 623, 2],
+    ]
+
+    const totals: number[] = []
+    for (let i = 0; i < levels.length; i += 1) {
+      totals.push(0)
+    }
```

Comment:
I never tried it but apparently Array has a `fill()` method, something like:

```javascript
const totals: number[] = new Array(levels.length).fill(0)
```

It may read better and maybe even be more efficient...I'm just throwing it as it's a nice trick more than to say you should change it :)

Reply:
yes, good idea. can i do that in the S3 PR? this file is changing a lot in the other branch and i'll have terrible merge conflicts as it is.

i'm coming across more and more places where i can generalise these functions

Reply:
Of course! As I said, was mainly just sharing a cool/concise way of doing it, don't have to change and it can certainly be done in a separate PR :)